settings screen with theme and feed sort/filter options

GroovinChip opened the pull request.

ThinkDigitalSoftware commented on lib/screens/home/home.dart:
```diff
@@ -67,9 +68,66 @@ class _HomeState extends State<Home> with TickerProviderStateMixin {
+      default:
+        break;
+    }
+    String filter = prefs.get('FeedFilterSelection');
```
This could be shrunk into a single setState call with the variable assignment inside once you figure out which variable, or you can put the switch statement inside it

GroovinChip commented:
- Addressed change request
- Removed back button interceptor from package details page
- Addressed some warnings
- Added back button interceptor for handling system back navigation from the Settings page to the packages feed. This is in order to make the user's changes to filtering and sorting the feed immediately available. The AppBar back button does not require this.

GroovinChip commented:
- Re-introduced back button interceptor for package details screen so that if a user is on any other page in the package feed than page 1, they will be taken back to that page instead of page 1
- Bumped version to 0.8.1
